src/hooks/useEvaluateFormFieldExpressions.ts

```diff
@@ -38,7 +40,8 @@ export const useEvaluateFormFieldExpressions = (
           });
         }
       } else {
-        field.isHidden = false;
+        field.isHidden =
+          hideUnansweredQuestionsInReadonlyForms && sessionMode === 'embedded-view' && !hasAnswer(field, formValues);
```

Comment:
What happens if the unanswered question has a `hideWhenExpression` expression? Handling this from this place seems like a mix of concerns; the [form-field renderer](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/main/src/components/renderer/field/form-field-renderer.component.tsx#L179-L181) is probably the right place for handling this.

Reply:
First, the logic checks whether the question has a `hideWhenExpression`. If it does, the evaluation of that expression determines the result, and the rest of the code is skipped. In simpler terms, the flow can be summarized as:
`if (hasHideExpression()) {
  return evaluateHideExpression();
} else {
  return configurationFlag && sessionModeEmbeddedView && !hasAnswer();
}`

Reply:
What happens if the `hideWhenExpression` returns false?

Reply:
@samuelmale I’ve switched to using the Field Renderer instead of the hook.